pkg/image/registry/imagestreamimport/rest.go

```diff
@@ -208,6 +209,18 @@ func (r *REST) Create(ctx apirequest.Context, obj runtime.Object, createValidati
 		return nil, kapierrors.NewInternalError(err)
 	}
 
+	// check imported images status. If we get authentication error (401), try import same image without authentication.
+	// Docker registry gives 401 on public images if you have wrong secret in your secret list.
+	for _, image := range isi.Status.Images {
+		if image.Status.Reason == metav1.StatusReasonUnauthorized && strings.Contains(image.Status.Message, "incorrect username or password") {
```

Comment:
If I dont check for the specific message this breaks 'TestImageStreamImportAuthenticated` test. I still need to understand which behavior of the test I break and if this is the change for the test or change for logic. For now, by adding this it makes the test pass and at the same time solves the bug.

Reply:
this sort of check is definitely fragile, i'd prefer to understand what might need to change about the test over adding this contains check.

Reply:
we can put this on hold for now as I need to find few days to investigate this more.

Reply:
i'd still like to revisit this, but it's an improvement for now so i'm going to merge.